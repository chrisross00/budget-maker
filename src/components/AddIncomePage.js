import React from 'react';
import { connect } from 'react-redux';
import { addIncome } from '../actions/income';
import IncomeForm from './IncomeForm';


export class AddIncomePage extends React.Component {
  onSubmit = (income) => {
    this.props.addIncome(income);
    this.props.history.push('/');

    console.log('hit parent onSubmit');
  }
  render() {
    return (
      <div>
        <h2>Enter your income</h2>
        <IncomeForm
          onSubmit={this.onSubmit} />
      </div>
    );
  };
};

const mapDispatchToProps = (dispatch) => ({
  addIncome: (income) => dispatch(addIncome(income))
});


export default connect(undefined, mapDispatchToProps)(AddIncomePage);
