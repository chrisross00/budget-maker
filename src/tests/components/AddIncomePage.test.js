import React from 'react';
import { shallow } from 'enzyme';
import AddIncomePage from '../../components/AddIncomePage';

test('should render AddIncomePage', () => {
  const wrapper = shallow(<AddIncomePage />);
  expect(wrapper).toMatchSnapshot();
});
