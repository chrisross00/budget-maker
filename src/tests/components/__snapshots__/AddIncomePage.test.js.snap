// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddIncomePage 1`] = `
<div>
  <h2>
    Enter your income
  </h2>
  <Connect(IncomeForm) />
</div>
`;
